cpu/stm32l1/include/periph_cpu.h

```diff
@@ -108,6 +108,19 @@ typedef struct {
 #define PERIPH_SPI_NEEDS_TRANSFER_REGS
 /** @} */
 
+/**
+ * @brief   I2C device configuration
+ */
+typedef struct {
+    I2C_TypeDef *dev;       /**< pointer to the used I2C device */
+    gpio_t scl_pin;         /**< pin used for SCL */
+    gpio_t sda_pin;         /**< pin used for SDA */
+    uint8_t af_scl_sda;     /**< SCL or SDA AF */
+    uint8_t ev_irqn;        /**< I2C device Event Interrupt */
+    uint8_t er_irqn;        /**< I2C device Error Interrupt */
+    uint32_t clk_en;        /**< Clock enable register */
```

Comment:
@BytesGalore Yes I already tried this, but I got an error like: the i2c_cfg is constant and the Mutex is not.